### PR TITLE
:sparkles: UIButton component (EPI-138/EPI-148)

### PR DESCRIPTION
**Pull request by AubaneNourry**
This pull request introduces the `IntField` and `UIButton` components to the project. It includes updates to the CMakeLists files, new class implementations, and modifications to existing engine and graphics code to integrate these components.

### New Components

* **IntField Component**:
  - Added `IntField.hpp` and `IntField.cpp` to define and implement the `IntField` class, which extends `AField` to handle integer field types. (`src/common/fields/IntField.hpp`, `src/common/fields/IntField.cpp`) [[1]](diffhunk://#diff-35d0d31fbfcc17bdbbfc00524206281fbb2decd2aa8fa43f3e3e6e28c40673dcR1-R39) [[2]](diffhunk://#diff-0a80509d4dd44f41a7968a09a5ccb531a65f6d01f7b33c962755107462a9b7c2R1-R48)
  - Updated `CMakeLists.txt` to include `IntField.hpp` and `IntField.cpp`. (`src/common/fields/CMakeLists.txt`)

* **UIButton Component**:
  - Added `UIButton.hpp` and `UIButton.cpp` to define and implement the `UIButton` class, which represents an interactive button component in the UI. (`src/graphics/components/UIButton.hpp`, `src/graphics/components/UIButton.cpp`) [[1]](diffhunk://#diff-9f5ed8417d742271631712e00031dc1d9ffc4c135ec390ecea96f88574c4bcacR1-R123) [[2]](diffhunk://#diff-55aff90d57716a2f2a91001f942dadc2b0f3cfdcf5918f2cef274a784a186869R1-R186)
  - Updated `CMakeLists.txt` to include `UIButton.hpp` and `UIButton.cpp`. (`src/graphics/components/CMakeLists.txt`)

### Engine and Graphics Integration

* **Engine Component Registration**:
  - Updated `Engine.cpp` to include and register the `UIButton` component. (`src/engine/Engine.cpp`) [[1]](diffhunk://#diff-792e7ee9541fb28c61cdf063f032057c64c788f41fe955bae6b5c1a702f2f15bR30) [[2]](diffhunk://#diff-792e7ee9541fb28c61cdf063f032057c64c788f41fe955bae6b5c1a702f2f15bR43)

* **Graphics Enhancements**:
  - Modified `Graphics.cpp` to handle null object pointers and integrate mouse event handling for `UIButton` interactions. (`src/graphics/Graphics.cpp`) [[1]](diffhunk://#diff-1f6629087aa105bc8673367f0a62763b3f9c2b2e562c19a6a6cf2a5c98c3233cR46-R48) [[2]](diffhunk://#diff-1f6629087aa105bc8673367f0a62763b3f9c2b2e562c19a6a6cf2a5c98c3233cR209-R214) [[3]](diffhunk://#diff-1f6629087aa105bc8673367f0a62763b3f9c2b2e562c19a6a6cf2a5c98c3233cL217) [[4]](diffhunk://#diff-1f6629087aa105bc8673367f0a62763b3f9c2b2e562c19a6a6cf2a5c98c3233cR240-R242) [[5]](diffhunk://#diff-1f6629087aa105bc8673367f0a62763b3f9c2b2e562c19a6a6cf2a5c98c3233cL259)

### CODE REVIEW

File: src/graphics/components/UIButton.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                                [this] { return this->_owner->_rectY; });
+    auto *field4 = new FloatField("width", "The width of the button",
+                                [this](const float &value) {
+                                    this->_owner->_width = value;
+                                },
+                                [this] { return this->_owner->_width; });
+    auto *field5 = new FloatField("height", "The height of the button",
+                                [this](const float &value) {
+                                    this->_owner->_height = value;
+                                },
```
Comment (RenardFute) [inline src/graphics/components/UIButton.cpp:99]: Can you use Vector2Field por favor ?

**Review by RenardFute — CHANGES_REQUESTED** [review]
Where is the code that draws the button ?

File: src/graphics/components/UIButton.cpp (hunk 1 above)
Comment (Marius-P1) [reply src/graphics/components/UIButton.cpp:99]: +1

**Review by Marius-P1 — APPROVED** [review]
👌

**Comment by AubaneNourry** [comment]
@RenardFute @Marius-P1 I added a Vector2Field and separate Button and UIButton for a better Developper experience

**Review by RenardFute — APPROVED** [review]

**Pull request merged**